llm/training/config.py

```diff
@@ -0,0 +1,341 @@
+import tempfile
+
+import fsspec.generic
+from datasets import load_dataset, load_from_disk, Dataset
+from transformers import TrainingArguments, TrainerCallback, TrainerState, TrainerControl
+from peft import LoraConfig
+import torch
+from saturnfs import SaturnFS
+import fsspec
+
+from llm.model_configs import ModelConfig
+from llm.prompt import Prompt, Role, Message, UserAssistantFormat, RedpajamaFormat, VicunaFormat, Llama2Format, \
+    TogetherLlama2Format, ChatMLFormat, DollyFormat, PromptFormat
+from llm.qa.prompts import ZeroShotQA, FewShotQA, StandaloneQuestion
+
+fsspec.register_implementation("sfs", SaturnFS)
+
+
+class ConfigError(Exception):
+    pass
+
+
+experiment_tracking_methods = {}
+
+
+@dataclass
+class ExperimentTrackingConfig:
+    """
+    This configuration object defines values pertaining to different
+    experiment tracking solutions. In many cases, this config
+    will be responsible for setting environment variables.
+
+    This isn't always necessary,most users can set configuration variables outside
+    of this config, but this exists to make it easy to configure experiment tracking in config.
+    As a result - most of the values will be optional
+    """
+
+    method: str
+    kwargs: Dict[str, Any] = field(default_factory=dict)
+    report_to: Optional[str] = None
+    # holds objects (like experiments) returned by the framework
+    objs: Dict[str, Any] = field(default_factory=dict)
+
+    def __post_init__(self):
+        self.objs = self.begin()
+
+    @classmethod
+    def register(cls, name, method) -> None:
+        experiment_tracking_methods[name] = method
+
+    def begin(self):
+        return experiment_tracking_methods[self.method](self, **self.kwargs)
+
+
+def init_comet_ml(
+    config: ExperimentTrackingConfig,
+    comet_api_key: Optional[str] = None,
+    comet_workspace: Optional[str] = None,
+    comet_project_name: Optional[str] = None,
+    auto_output_logging=None,
+) -> Dict[str, Any]:
+    import comet_ml
+    comet_ml.init(
+        api_key=comet_api_key or os.getenv("COMET_API_KEY"),
+        workspace=comet_workspace or os.getenv("COMET_WORKSPACE"),
+        project_name=comet_project_name or os.getenv("COMET_PROJECT_NAME"),
+    )
+    os.environ["COMET_LOG_ASSETS"] = "True"
+    config.report_to = "comet_ml"
+    return {}
+
+
+def init_wandb(
+    config: ExperimentTrackingConfig,
+    wandb_project: Optional[str] = None,
+    wandb_log_model: str = "all",
+) -> Dict[str, Any]:
+    if wandb_project:
+        os.environ["WANDB_PROJECT"] = wandb_project
+    if wandb_log_model:
+        os.environ["WANDB_LOG_MODEL"] = wandb_log_model
+    config.report_to = "wandb"
+    return {}
+
+ExperimentTrackingConfig.register("comet_ml", init_comet_ml)
+ExperimentTrackingConfig.register("wandb", init_wandb)
+
+
+dataset_method_registry = {}
+
+
+@dataclass
+class DatasetConfig:
+    # name or path to dataset
+    method: str
+    kwargs: Dict[str, Any]
```

Comment:
Would it make sense to have `load_dataset` be the default method?

Also wondering if it will get frustrating having to remember that `load_dataset` needs the kwarg "path" and `load_from_disk` needs the kwarg "dataset_path"